test_automation/work_order_utility.py

```diff
@@ -0,0 +1,171 @@
+    """ Function to process work order get result response. """
+
+    # process work order get result and retrieve response
+    input_json_str1 = create_work_order_get_result(work_order_id)
+    logger.info("------------------Validating WorkOrderGetResult Response------------------")
+    response = {}
+    output_json_file_name = 'work_order_get_result'
+
+    while("result" not in response):
+        if "error" in response:
+            if response["error"]["code"] != 5:
+                logger.info("ERROR: Failed at WorkOrderGetResult - Response received with error code.")
+                err_cd = 1
+                break
+
+        response = process_request(uri_client, input_json_str1, output_json_file_name)
+        time.sleep(3)
+
+    return response
+
+def process_work_order(input_json_file, output_json_file_name, worker_obj, sig_obj, uri_client, private_key, err_cd, check_submit, check_get_result):
+    """ Function to process work order and perform basic validation of return code in response. """
+
+    if err_cd == 0:
+        # read json input file for the test case
+        logger.info("------------------Input file name: %s ---------------\n", input_json_file)
+        input_json_str1 = futils.read_json_file(input_json_file)
+        #----------------------------------------------------------------------------------
+        logger.info("------------------Testing WorkOrderSubmit------------------")
+
+        # create work order request
+        input_json_str1, work_order_id = create_work_order_request(input_json_str1, worker_obj)
+        # sign work order request
+        input_json_str1, session_iv, encrypted_session_key = sign_work_order_request(input_json_str1, worker_obj, sig_obj, private_key)
+        # submit work order request and retrieve response
+        response = process_request(uri_client, input_json_str1, output_json_file_name)
+        # validate work order response and get error code
+        err_cd = validate_work_order_response(response, check_submit)
+
+        if err_cd == 0:
+            logger.info("------------------Testing WorkOrderGetResult------------------")
+            # submit work order get result request and retrieve response
+            response = process_work_order_get_result(work_order_id, uri_client)
+            # check expected key of test case
+            key_name = list(check_get_result.keys())[0]
+            logger.info("key_name in expected result: %s  \n", key_name)
+            # validate response result or error codes
+            if key_name in response:
+                if "code" in check_get_result[key_name].keys():
+                    if "code" in response[key_name].keys():
+                        if response[key_name]["code"] == check_get_result[key_name]["code"]:
+                            err_cd = 0
+                            logger.info("SUCCESS: WorkOrderGetResult Processed Successfully and contains expected result code. First Check.")
```

Comment:
This line is too long and is hard to read. Please limit lines to 80 characters.

Reply:
Fixed line length to below 80